sale_order_lot_selection/model/sale.py

```diff
@@ -72,6 +72,7 @@ def _check_move_state(self, line):
     @api.multi
     def action_confirm(self):
         res = super(SaleOrder, self).action_confirm()
-        for line in self.order_line:
-            self._check_move_state(line)
+        if res is True:
+            for line in self.order_line:
```

Comment:
self may contain multiple records.
So, always loop order the records i.e. for order in self: